---
hide_table_of_contents: true
hide_title: true
custom_edit_url: null
---

import CenterLayout from "/src/components/CenterLayout"
import GitHub from "/src/components/GitHub"

<CenterLayout>
<span className="searchCategory">Knowledge Base</span>
<h1>Re-connecting to OpenVPN Instance after destroying and re-applying it</h1>
<GitHub discussion={{"id":"D_kwDOF8slf84AN3Mi","number":38,"author":{"login":"zackproser"},"title":"Re-connecting to OpenVPN Instance after destroying and re-applying it","body":"A customer asked: \r\n> Hello, we’re having an issue with our Openvpn server in our stage environment. Our environment was set up originally with the Gruntwork Reference Architecture.  We got a notice from AWS about a detected degradation of the underlying hardware hosting your Amazon EC2 instance. So we used Terraform to destroy the openvpn server module and then recreated it (I’m guessing this wasn’t the right way to do it). We use Viscocity to connect to the openvpn servers. I changed the IP for the existing connection to the new public IP for the new openvpn server but I cannot connect. Below is snippet from  my Viscocity log when trying to connect (certain information has been X’d out for security):\r\n```\r\n2021-10-15 14:29:29: Viscosity Mac 1.9.4 (1578)\r\n2021-10-15 14:29:29: Viscosity OpenVPN Engine Started\r\n2021-10-15 14:29:29: Running on macOS 11.6.0\r\n2021-10-15 14:29:29: ---------\r\n2021-10-15 14:29:29: State changed to Connecting\r\n2021-10-15 14:29:29: Checking reachability status of connection...\r\n2021-10-15 14:29:29: Connection is reachable. Starting connection attempt.\r\n2021-10-15 14:29:29: OpenVPN 2.4.11 x86_64-apple-darwin [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [MH/RECVDA] [AEAD] built on Aug 26 2021\r\n2021-10-15 14:29:29: library versions: OpenSSL 1.1.1l  24 Aug 2021, LZO 2.10\r\n2021-10-15 14:29:29: Valid endpoint found: XX.X.XXX.205:1194:udp\r\n2021-10-15 14:29:29: TCP/UDP: Preserving recently used remote address: [AF_INET]XX.X.XXX.205:1194\r\n2021-10-15 14:29:29: UDP link local: (not bound)\r\n2021-10-15 14:29:29: UDP link remote: [AF_INET]52.7.152.205:1194\r\n2021-10-15 14:29:29: State changed to Authenticating\r\n2021-10-15 14:29:29: VERIFY ERROR: depth=1, error=self signed certificate in certificate chain: C=US, ST=ME, L=Farmington, O=Aptuitiv, OU=IT, CN=Aptuitiv CA, name=server, emailAddress=XXXX@XXXX.com, serial=16374791465934600503\r\n2021-10-15 14:29:29: OpenSSL: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed\r\n2021-10-15 14:29:29: TLS_ERROR: BIO read tls_read_plaintext error\r\n2021-10-15 14:29:29: TLS Error: TLS object -> incoming plaintext read error\r\n2021-10-15 14:29:29: TLS Error: TLS handshake failed\r\n2021-10-15 14:29:29: SIGTERM[soft,tls-error] received, process exiting\r\n2021-10-15 14:29:29: State changed to Disconnected (Process Terminated)\r\n```","bodyHTML":"<p dir=\"auto\">A customer asked:</p>\n<blockquote>\n<p dir=\"auto\">Hello, we’re having an issue with our Openvpn server in our stage environment. Our environment was set up originally with the Gruntwork Reference Architecture.  We got a notice from AWS about a detected degradation of the underlying hardware hosting your Amazon EC2 instance. So we used Terraform to destroy the openvpn server module and then recreated it (I’m guessing this wasn’t the right way to do it). We use Viscocity to connect to the openvpn servers. I changed the IP for the existing connection to the new public IP for the new openvpn server but I cannot connect. Below is snippet from  my Viscocity log when trying to connect (certain information has been X’d out for security):</p>\n</blockquote>\n<div class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"2021-10-15 14:29:29: Viscosity Mac 1.9.4 (1578)\n2021-10-15 14:29:29: Viscosity OpenVPN Engine Started\n2021-10-15 14:29:29: Running on macOS 11.6.0\n2021-10-15 14:29:29: ---------\n2021-10-15 14:29:29: State changed to Connecting\n2021-10-15 14:29:29: Checking reachability status of connection...\n2021-10-15 14:29:29: Connection is reachable. Starting connection attempt.\n2021-10-15 14:29:29: OpenVPN 2.4.11 x86_64-apple-darwin [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [MH/RECVDA] [AEAD] built on Aug 26 2021\n2021-10-15 14:29:29: library versions: OpenSSL 1.1.1l  24 Aug 2021, LZO 2.10\n2021-10-15 14:29:29: Valid endpoint found: XX.X.XXX.205:1194:udp\n2021-10-15 14:29:29: TCP/UDP: Preserving recently used remote address: [AF_INET]XX.X.XXX.205:1194\n2021-10-15 14:29:29: UDP link local: (not bound)\n2021-10-15 14:29:29: UDP link remote: [AF_INET]52.7.152.205:1194\n2021-10-15 14:29:29: State changed to Authenticating\n2021-10-15 14:29:29: VERIFY ERROR: depth=1, error=self signed certificate in certificate chain: C=US, ST=ME, L=Farmington, O=Aptuitiv, OU=IT, CN=Aptuitiv CA, name=server, emailAddress=XXXX@XXXX.com, serial=16374791465934600503\n2021-10-15 14:29:29: OpenSSL: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed\n2021-10-15 14:29:29: TLS_ERROR: BIO read tls_read_plaintext error\n2021-10-15 14:29:29: TLS Error: TLS object -&gt; incoming plaintext read error\n2021-10-15 14:29:29: TLS Error: TLS handshake failed\n2021-10-15 14:29:29: SIGTERM[soft,tls-error] received, process exiting\n2021-10-15 14:29:29: State changed to Disconnected (Process Terminated)\"><pre class=\"notranslate\"><code>2021-10-15 14:29:29: Viscosity Mac 1.9.4 (1578)\n2021-10-15 14:29:29: Viscosity OpenVPN Engine Started\n2021-10-15 14:29:29: Running on macOS 11.6.0\n2021-10-15 14:29:29: ---------\n2021-10-15 14:29:29: State changed to Connecting\n2021-10-15 14:29:29: Checking reachability status of connection...\n2021-10-15 14:29:29: Connection is reachable. Starting connection attempt.\n2021-10-15 14:29:29: OpenVPN 2.4.11 x86_64-apple-darwin [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [MH/RECVDA] [AEAD] built on Aug 26 2021\n2021-10-15 14:29:29: library versions: OpenSSL 1.1.1l  24 Aug 2021, LZO 2.10\n2021-10-15 14:29:29: Valid endpoint found: XX.X.XXX.205:1194:udp\n2021-10-15 14:29:29: TCP/UDP: Preserving recently used remote address: [AF_INET]XX.X.XXX.205:1194\n2021-10-15 14:29:29: UDP link local: (not bound)\n2021-10-15 14:29:29: UDP link remote: [AF_INET]52.7.152.205:1194\n2021-10-15 14:29:29: State changed to Authenticating\n2021-10-15 14:29:29: VERIFY ERROR: depth=1, error=self signed certificate in certificate chain: C=US, ST=ME, L=Farmington, O=Aptuitiv, OU=IT, CN=Aptuitiv CA, name=server, emailAddress=XXXX@XXXX.com, serial=16374791465934600503\n2021-10-15 14:29:29: OpenSSL: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed\n2021-10-15 14:29:29: TLS_ERROR: BIO read tls_read_plaintext error\n2021-10-15 14:29:29: TLS Error: TLS object -&gt; incoming plaintext read error\n2021-10-15 14:29:29: TLS Error: TLS handshake failed\n2021-10-15 14:29:29: SIGTERM[soft,tls-error] received, process exiting\n2021-10-15 14:29:29: State changed to Disconnected (Process Terminated)\n</code></pre></div>","answer":{"body":"The right thing to do in this case is terminate the current VPN server (perform an instance refresh on the Auto Scaling Group (ASG) that the openvpn server is attached to). \r\n\r\nThe new host will automatically grab the same Elastic IP address (EIP) and will restore the same certificates. ","bodyHTML":"<p dir=\"auto\">The right thing to do in this case is terminate the current VPN server (perform an instance refresh on the Auto Scaling Group (ASG) that the openvpn server is attached to).</p>\n<p dir=\"auto\">The new host will automatically grab the same Elastic IP address (EIP) and will restore the same certificates.</p>"}}} />

</CenterLayout>
  

<!-- ##DOCS-SOURCER-START
{
  "sourcePlugin": "github-discussions",
  "hash": "76f6b66e00763d9b16093246891f5cac"
}
##DOCS-SOURCER-END -->
